src/Text/Trifecta/Rope.hs

```diff
@@ -59,22 +78,69 @@ data Rope = Rope !Delta !(FingerTree Delta Strand) deriving Show
 rope :: FingerTree Delta Strand -> Rope
 rope r = Rope (measure r) r
 
+-- | Construct a 'Rope' out of a single 'ByteString' strand.
+ropeBS :: ByteString -> Rope
+ropeBS = rope . singleton . strand
+
 strands :: Rope -> FingerTree Delta Strand
 strands (Rope _ r) = r
 
--- | grab a the contents of a rope from a given location up to a newline
-grabRest :: Delta -> Rope -> r -> (Delta -> Lazy.ByteString -> r) -> r
-grabRest i t kf ks = trim (delta l) (bytes i - bytes l) (toList r) where
-  trim j 0 (Strand h _ : xs) = go j h xs
-  trim _ k (Strand h _ : xs) = go i (Strict.drop (fromIntegral k) h) xs
-  trim j k (p          : xs) = trim (j <> delta p) k xs
-  trim _ _ []                = kf
-  go j h s = ks j $ Lazy.fromChunks $ h : [ a | Strand a _ <- s ]
-  (l, r) = FingerTree.split (\b -> bytes b > bytes i) $ strands t
-
--- | grab a the contents of a rope from a given location up to a newline
-grabLine :: Delta -> Rope -> r -> (Delta -> Strict.ByteString -> r) -> r
-grabLine i t kf ks = grabRest i t kf $ \c -> ks c . Util.fromLazy . Util.takeLine
+-- | Grab the entire rest of the input 'Rope', starting at an initial offset, or
+-- return a default if we’re already at or beyond the end. Also see 'grabLine'.
+--
+-- Extract a suffix of a certain length from the input:
+--
+-- >>> grabRest (delta ("Hello " :: ByteString)) (ropeBS "Hello World\nLorem") Nothing (\x y -> Just (x, Lazy.toString y))
```

Comment:
This could be much easier using `String`s instead of `ByteString`s for the input:

```haskell
grabRest (delta "Hello ") (unit "Hello World\n Lorem") Nothing (\x y -> Just (x, Lazy.toString y))
```
(Without `-XOverloadedStrings`, we don't need the type annotation either.)